src/sql/src/plan/error.rs

```diff
@@ -231,6 +232,23 @@ pub enum PlanError {
     // TODO(benesch): eventually all errors should be structured.
     Unstructured(String),
 }
+#[derive(Clone, Debug)]
+pub enum ConnectionParsingError {
+    MissingAwsCredentials,
+    MissingAssumeRoleArn,
+    ConflictingOptions,
+    MissingRequiredOptions,
+}
+impl fmt::Display for ConnectionParsingError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::MissingAwsCredentials => f.write_str("invalid CONNECTION: must specify both ACCESS KEY ID and SECRET ACCESS KEY with optional TOKEN"),
+            Self::MissingAssumeRoleArn => f.write_str("invalid CONNECTION: must specify ASSUME ROLE ARN with optional ASSUME ROLE SESSION NAME"),
+            Self::ConflictingOptions => f.write_str("invalid CONNECTION: ASSUME ROLE ARN cannot be provided simultaneously with ACCESS KEY ID and SECRET ACCESS KEY"),
+            Self::MissingRequiredOptions => f.write_str("invalid CONNECTION: must specify either ASSUME ROLE ARN or both ACCESS KEY ID and SECRET ACCESS KEY"),
```

Comment:
These two options have AWS specific error messages but the variants don't indicate that they are AWS specific.